packages/ui-chart-components/src/components/Reference/Legend.tsx

```diff
@@ -97,33 +102,50 @@ const Text = styled.span`
   line-height: 1.4;
 `;
 
-const getPieLegendDisplayValue = (chartConfig, value, item, isEnlarged, viewContent) => {
-  if (chartConfig[value]?.label) {
-    return chartConfig[value].label;
+function isPieChartConfig(config: PieChartConfig | {}): config is PieChartConfig {
+  return (config as PieChartConfig).chartType === 'pie';
+}
+
+const getPieLegendDisplayValue = (
+  chartConfig: PieChartConfig | {},
+  value: string,
+  item: any,
+  viewContent: ViewContent,
+  isEnlarged?: boolean,
+) => {
+  if (isPieChartConfig(chartConfig) && chartConfig[value as keyof PieChartConfig]?.label) {
+    return chartConfig[value as keyof PieChartConfig].label;
   }
-
   const metadata = item[`${value}_metadata`];
   const labelSuffix = formatDataValueByType({ value: item.value, metadata }, viewContent.valueType);
 
   // on mobile the legend will show the actual formatDataValueByType after the label value
   return isMobile() && isEnlarged ? `${value} ${labelSuffix}` : value;
 };
 
+interface PieLegendProps {
+  chartConfig: PieChartConfig | {};
+  isEnlarged?: boolean;
+  isExporting?: boolean;
+  legendPosition?: LegendPosition;
+  viewContent: ViewContent;
+}
+
 export const getPieLegend = ({
   chartConfig = {},
   isEnlarged,
   isExporting,
   legendPosition,
   viewContent,
-}) => ({ payload }) => (
+}: PieLegendProps) => ({ payload }: any) => (
```

Comment:
I had quite a bit of trouble with these functions because of the way they are a function inside a function. I have had good luck with adding a recharts type to the inside prop but when I add one to the outside prop I got a bunch of errors in difference files popping up.